android/src/main/java/com/adyenreactnativesdk/cse/ActionFragment.kt

```diff
@@ -0,0 +1,99 @@
+package com.adyenreactnativesdk.cse;
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.adyen.checkout.action.core.GenericActionComponent
+import com.adyen.checkout.action.core.GenericActionConfiguration
+import com.adyen.checkout.components.core.ActionComponentCallback
+import com.adyen.checkout.components.core.ActionComponentData
+import com.adyen.checkout.components.core.action.Action
+import com.adyen.checkout.ui.core.AdyenComponentView
+import com.adyenreactnativesdk.AdyenCheckout
+import com.adyenreactnativesdk.R
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import kotlinx.coroutines.launch
+
+class ActionFragment(
+    private val configuration: GenericActionConfiguration,
+    private val callback: ActionComponentCallback,
+    private val action: Action,
+) :
+    BottomSheetDialogFragment() {
+
+    private var actionHandled: Boolean = false
+    var component: GenericActionComponent? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        dialog?.setCanceledOnTouchOutside(false)
+        return inflater.inflate(R.layout.fragment_instant, container)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
```

Comment:
Let's discuss if we can use the onStart() callback of the fragment instead of the onViewCreated. Maybe we can just call setupComponent() in the onStart() callback.

Reply:
Fixed [eb8d84c](https://github.com/Adyen/adyen-react-native/pull/338/commits/eb8d84c3cf6a325641d6cd41576f76bb762b1c87)